basic sharepoint UX in docs (#388)
spiceai/docs: spiceaidocs/docs/components/data-connectors/sharepoint.md

---
title: 'Sharepoint Data Connector'
sidebar_label: 'Sharepoint Data Connector'
description: 'Sharepoint Data Connector Documentation'
pagination_prev: null
---

The Sharepoint Data Connector enables federated SQL queries on documents stored in Sharepoint.

```yaml
datasets:
  - from: sharepoint:drive:Documents/path:/nuclear_secrets/
    name: important_documents
    params:
      sharepoint_client_id: ${secrets:SPICE_SHAREPOINT_CLIENT_ID}
      sharepoint_tenant_id: ${secrets:SPICE_SHAREPOINT_TENANT_ID}
      sharepoint_client_secret: ${secrets:SPICE_SHAREPOINT_CLIENT_SECRET}
```
#### Example
```sql
SELECT * FROM important_documents limit 1
```
Returns
```json

[
    {
        "created_by_id": "cbccd193-f9f1-4603-b01d-ff6f3e6f2108",
        "created_by_name": "Jack Eadie",

>         "created_date_time": "2024-09-09T04:57:00",

Reviewer: ```suggestion
        "created_at": "2024-09-09T04:57:00",
```

        "c_tag": "\"c:{BD4D130F-2C95-4E59-9F93-85BD0A9E1B19},1\"",
        "e_tag": "\"{BD4D130F-2C95-4E59-9F93-85BD0A9E1B19},1\"",
        "id": "01YRH3MPAPCNG33FJMLFHJ7E4FXUFJ4GYZ",
        "last_modified_by_id": "cbccd193-f9f1-4603-b01d-ff6f3e6f2108",
        "last_modified_by_name": "Jack Eadie",

>         "last_modified_date_time": "2024-09-09T04:57:00",

Reviewer: ```suggestion
        "last_modified_at": "2024-09-09T04:57:00",
```

    }
]
```

:::warning[Limitations]
The sharepoint connector does not yet support creating a dataset from a single file (e.g. an Excel spreadsheet). Datasets must be created from a folder of documents (see [Document Support](/components/data-connectors/index.md#document-support)).
:::

## Configuration
### Parameters

- `sharepoint_client_id`: Required. The client ID of the Azure AD (Entra) application.
- `sharepoint_tenant_id`: Required. The tenant ID of the Azure AD (Entra) application.
- `sharepoint_client_secret`: Optional. For service principal authentication. The client secret of the Azure AD (Entra) application.
- `sharepoint_auth_code`: Optional. For user authentication. The authorization code obtained from the OAuth2 flow (see `spice login sharepoint` [docs](/cli/reference/login)).

> Note: Only one of `sharepoint_client_secret` or `sharepoint_authorization_code` is allowed.

Reviewer: ```suggestion
Note: Only one of `sharepoint_client_secret` or `sharepoint_auth_code` is allowed.
```

### Creating an Enterprise Application
To use the Sharepoint connector with service principal authentication, you will need to create an Azure AD application and grant it the necessary permissions. This will also support OAuth2 authentication for users within the tenant (i.e. `params.sharepoint_auth_code`).

1. Create a new Azure AD application in the [Azure portal](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/Overview).
2. Under the application's `API permissions`, add the following permissions: `Sites.Read.All`, `Files.Read.All`, `User.Read`, `GroupMember.Read.All`
  - For service principal authentication, Application permissions are required. For user authentication, only delegated permissions are required.
3. Add `sharepoint_client_id` and `sharepoint_tenant_id` to the connector configuration.
4. (For service principal authentication): Under the application's `Certificates & secrets`, create a new client secret. Use this in `params.sharepoint_client_secret`.

### Default Spice Application

> For convenience, the SpiceAI team maintains a default Azure AD application that can be used for Sharepoint authentication. This application requires OAuth2 authentication. To use it:

Reviewer: ```suggestion
For convenience, Spice AI maintains a default Azure AD application that can be used for Sharepoint authentication. This application requires OAuth2 authentication. To use it:
```

---
title: 'Sharepoint Data Connector'
sidebar_label: 'Sharepoint Data Connector'
description: 'Sharepoint Data Connector Documentation'
pagination_prev: null
---

The Sharepoint Data Connector enables federated SQL queries on documents stored in Sharepoint.

```yaml
datasets:
  - from: sharepoint:drive:Documents/path:/nuclear_secrets/
    name: important_documents
    params:
      sharepoint_client_id: ${secrets:SPICE_SHAREPOINT_CLIENT_ID}
      sharepoint_tenant_id: ${secrets:SPICE_SHAREPOINT_TENANT_ID}
      sharepoint_client_secret: ${secrets:SPICE_SHAREPOINT_CLIENT_SECRET}
```

:::warning[Limitations]

> The sharepoint connector does not support creating a dataset from a single file (e.g. an Excel spreadsheet). Datasets must be created from a folder of documents.

Reviewer: ```suggestion
The sharepoint connector does not yet support creating a dataset from a single file (e.g. an Excel spreadsheet). Datasets must be created from a folder of documents.
```

> :::

Reviewer: Let's add some example query and results.

## Configuration
### Parameters

- `sharepoint_client_id`: Required. The client ID of the Azure AD (Entra) application.
- `sharepoint_tenant_id`: Required. The tenant ID of the Azure AD (Entra) application.
- `sharepoint_client_secret`: Optional. For service principal authentication. The client secret of the Azure AD (Entra) application.

> - `sharepoint_authorization_code`: Optional. For user authentication. The authorization code obtained from the OAuth2 flow (see `spice login sharepoint` [docs](cli/reference/login)).

Reviewer: ```suggestion
- `sharepoint_auth_code`: Optional. For user authentication. The authorization code obtained from the OAuth2 flow (see `spice login sharepoint` [docs](cli/reference/login)).
```

Note: Only one of `sharepoint_client_secret` or `sharepoint_authorization_code` is allowed.

### `from` formats

The `from` field in a Sharepoint dataset takes the following format:
```yaml
from: 'sharepoint:<drive_type>:<drive_id>/<subpath_type>:<subpath_value>'

> ```

Reviewer: Missing `file_format` and description about that?

Author: Because we only support documents, we don't inherently need `file_format`. We could use it, and apply it as a filter to files in a folder
